### PR TITLE
Changing field 'availale' depending on endpoint

### PR DESCRIPTION
**Pull request by bokimilinkovic**
https://trello.com/c/TscF6Er6/135-set-the-available-field-to-true-false-when-book-is-returned-borrowed

### CODE REVIEW

File: interactor/borrow_return.go [hunk 1]
```diff
@@ -28,6 +33,15 @@ func (l *LoanWriter) Borrow(userID int, bookID int) error {
 		return err
 	}
 
+	book, err := l.store.FindBookById(bookID)
```
Comment (sfqi) [inline interactor/borrow_return.go]: Don't we have a book loader middleware?

Comment (bokimilinkovic) [reply interactor/borrow_return.go]: yees, i forgot that we have book in context

Comment (bokimilinkovic) [reply interactor/borrow_return.go]: but if I get book from context, in loanhandler, I need to call UpdateBook directly there, or I can pass somehow that book in  loanWriter Interactor and than from there to call UpdateBook ? Which one is better ? @sfqi

Comment (sfqi) [reply interactor/borrow_return.go]: you pass the book to the interactor

File: handler/loanhandler.go [hunk 2]
```diff
@@ -114,11 +115,11 @@ func (wl *WriteLoanHandler) BorrowBook(w http.ResponseWriter, r *http.Request) *
 }
 
 func (wl *WriteLoanHandler) ReturnBook(w http.ResponseWriter, r *http.Request) *HTTPError {
-	id, err := strconv.Atoi(mux.Vars(r)["id"])
-	if err != nil {
-		return newHTTPError(http.StatusNotFound, err)
+	book := r.Context().Value("book")
```
Comment (sfqi) [inline handler/loanhandler.go]: you need to check the `ok` as well, otherwise it might panic

**Comment by sfqi** [comment]
@bokimilinkovic See if you can use git revert to undo this commit: https://github.com/sfqi/library/pull/97/commits/2afe6e17ccc20a9dd9e23038b480209b0610dc50

We won't be using the book from context

**Pull request merged**